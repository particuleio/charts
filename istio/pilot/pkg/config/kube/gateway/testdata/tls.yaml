apiVersion: networking.x-k8s.io/v1alpha1
kind: GatewayClass
metadata:
  name: istio
spec:
  controller: istio.io/gateway-controller
---
apiVersion: networking.x-k8s.io/v1alpha1
kind: Gateway
metadata:
  name: gateway
  namespace: default
spec:
  gatewayClassName: istio
  listeners:
  # Passthrough
  - port: 34000
    protocol: TLS
    routes:
      namespaces:
        from: All
      selector:
        matchLabels:
          selected: "yes"
      kind: TLSRoute
    tls:
      mode: Passthrough
      certificateRef:
        name: my-cert-tls
        group: core
        kind: Secret
  # Termination
  - hostname: "domain.example"
    port: 34000
    protocol: HTTPS
    routes:
      namespaces:
        from: All
      selector:
        matchLabels:
          selected: "yes"
      kind: HTTPRoute
    tls:
      mode: Terminate
      certificateRef:
        name: my-cert-http
        group: core
        kind: Secret
---
apiVersion: networking.x-k8s.io/v1alpha1
kind: TLSRoute
metadata:
  name: tls
  namespace: default
  labels:
    selected: "yes"
spec:
  rules:
  - forwardTo:
    - serviceName: httpbin
      port: 443
---
apiVersion: networking.x-k8s.io/v1alpha1
kind: TLSRoute
metadata:
  name: tls-match
  namespace: default
  labels:
    selected: "yes"
spec:
  rules:
  - matches:
    - snis: ["foo.com"]
    forwardTo:
    - serviceName: httpbin
      port: 443
---
apiVersion: networking.x-k8s.io/v1alpha1
kind: HTTPRoute
metadata:
  name: http
  namespace: default
  labels:
    selected: "yes"
spec:
  hostnames: ["domain.example"]
  rules:
  - forwardTo:
    - serviceName: httpbin
      port: 80