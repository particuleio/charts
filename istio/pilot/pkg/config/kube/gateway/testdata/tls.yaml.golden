apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  creationTimestamp: null
  name: gateway-istio-autogenerated-k8s-gateway
  namespace: default
spec:
  selector:
    istio: ingressgateway
  servers:
  - hosts:
    - '*'
    port:
      name: tls-34000-gateway-gateway-default
      number: 34000
      protocol: TLS
    tls: {}
  - hosts:
    - domain.example
    port:
      name: https-34000-gateway-gateway-default
      number: 34000
      protocol: HTTPS
    tls:
      credentialName: my-cert-http
      mode: SIMPLE
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  creationTimestamp: null
  name: tls-tls-istio-autogenerated-k8s-gateway
  namespace: default
spec:
  gateways:
  - default/gateway-istio-autogenerated-k8s-gateway
  hosts:
  - '*'
  tls:
  - match:
    - sniHosts:
      - '*'
    route:
    - destination:
        host: httpbin.default.svc.domain.suffix
        port:
          number: 443
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  creationTimestamp: null
  name: tls-match-tls-istio-autogenerated-k8s-gateway
  namespace: default
spec:
  gateways:
  - default/gateway-istio-autogenerated-k8s-gateway
  hosts:
  - '*'
  tls:
  - match:
    - sniHosts:
      - foo.com
    route:
    - destination:
        host: httpbin.default.svc.domain.suffix
        port:
          number: 443
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  creationTimestamp: null
  name: http-istio-autogenerated-k8s-gateway
  namespace: default
spec:
  gateways:
  - default/gateway-istio-autogenerated-k8s-gateway
  hosts:
  - domain.example
  http:
  - route:
    - destination:
        host: httpbin.default.svc.domain.suffix
        port:
          number: 80
---
